### ISSUE TITLE
Kernel Restarting The kernel keeps dying in a specific line

### ISSUE THREAD

**Issue by smajida** [ipython/ipython#8589]
Hi 
I'd like to implement a image descriptor called SIFT using import cv2. the program is quite easy and just a couple of line, but whenever I do shift+enter in a specific line below:
    
    skp = detector.detect(img)

I receive a message concerning: 

    Kernel Restarting
    The kernel appears to have died. It will restart automatically.

that's really strange because I am sure the program doen't have any problems.

I put the complete program:

     import cv2
     img = cv2.imread('/home/<user>/Pictures/galaxy.jpg')
     img = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)

     detector = cv2.FeatureDetector_create("SIFT")
     descriptor = cv2.DescriptorExtractor_create("SIFT")

      skp = detector.detect(img)
       skp, sd = descriptor.compute(img, skp)

which is line skp = detector.detect(img) the kernel of ipython dies.
ubuntu 14.04 lts
opencv 3.0.0
ipython 3.0.0
python 2.7.9
anaconda 2.2. x64bit
Cuda 7.0

**Comment by DavidJepherson** [comment]
I found a way around, but not a good way. I moved that notebook to another folder which it was the only file in that folder, ran it again and got no error. the bad side is that I need this notebook to be in a specific folder! Really Strange problem.

**Comment by minrk** [comment]
@DavidJepherson do you see any errors in the terminal where the notebook server is running? Things like this are typically C-level linking or runtime errors. Can you also run the same code in a script outside IPython?

**Comment by DavidJepherson** [comment]
@minrk I see this message in the terminal:
    2015-06-29 19:58:32.537 [NotebookApp] KernelRestarter: restarting kernel (1/5)
    WARNING:root:kernel 2b4b07f5-a856-4b95-94c2-bad74652d9ac restarted
    WARNING: Attempting to work in a virtualenv. If you encounter problems, please install IPython inside the virtualenv.

I had already intalled ipython in virtualenv, and tried again, but didn't work out.

in python script outsided ipython I got this error:
    Segmentation fault (core dumped)

**Comment by rgbkrk** [comment]
On your terminal, if the IPython in the virtualenv isn't getting picked up try running `hash -r` (assuming this is a bash terminal).

**Comment by minrk** [comment]
@DavidJepherson that message means a virtualenv is active and IPython is not installed in it. Is the IPython kernel in a virutalenv? Is the notebook server? What version of IPython are you using?

**Comment by DavidJepherson** [comment]
@rgbkrk 
I did so I got error like running outside ipython environment i.e like the message got in python
   Segmentation fault (core dumped)

**Comment by DavidJepherson** [comment]
@minrk 
I've a few times installed ipython in virtualenv to make sure it's installed. I installed ipython outside virtualenv after deactivating.  got this error when trying to run that line:

[I 10:49:31.427 NotebookApp] KernelRestarter: 
restarting kernel (1/5)
WARNING:root:kernel 9e6dc5df-0b2d-45dc-a8a6-b63ae7915b60 restarted

**Comment by takluyver** [comment]
Closing as probably not an IPython bug - it looks like something in cv2 is segfaulting. Feel free to use comments here to troubleshoot anyway.

**Comment by szsmith** [comment]
Has this problem been solved? Please point me to the updated solution. Thanks

**Comment by minrk** [comment]
@szsmith this problem turned out to be a C-level error in the library code being run, and independent of IPython. There is no solution for IPython, because IPython is unrelated. It will depend on what library is crashing, and there are many causes of C-level errors. Sometimes the fix for such errors is to reinstall the package that's segfaulting.

**Comment by nevion** [comment]
Looks like embed_kernel needs to support nesting, kind of like the original embed fixes I did.  Same save/restore trick to the singleton infrastructure should get that part working but... more unsure about the parent poller threads the IPKernelApp instance - shouldn't it share that instance across embeds so the active one dequeue the parents?

**Comment by minrk** [comment]
Yeah, it may need a bit of help to nest. It should use a global instance for the process, and on call set the namespace and resume the eventloop, reusing existing kernel/socket instances.

**Comment by GOPALSHEKHAWAT** [comment]
what is the purpose of interrupting and restarting the kernel,which is one the options in the toolbar in IPYTHON NOTEBOOK?

**Comment by takluyver** [comment]
Interrupting: if a computation is taking too long, you can break out of it without losing all of your variables.

Restarting: start again with a clean kernel state (remove all variables)

**Comment by CamiloGarcia22** [comment]
Hi, recently I installed de r-poRe package. The installation wasn't completely straightforward as I had to uninstall several packages from Anaconda (r-clipr, r-jdbc, r-reprex(with r-essetials), r-profvis. r-odbc and r-mongolite). Once the installation was successfully completed I realized that my Jupyter R Kernel was dead. the KerenelRestarter gives me this message:

[I 19:37:16.496 NotebookApp] KernelRestarter: restarting kernel (1/5)
dyld: Library not loaded: @rpath/libintl.8.dylib

Can someone help me with this issue?
